logprep/processor/amides/processor.py

```diff
@@ -218,15 +218,16 @@ def _apply_rules(self, event: dict, rule: AmidesRule):
         result = self._evaluate_cmdline_cached(normalized)
         self._update_cache_metrics()
 
-        if result:
-            self._write_target_field(event=event, rule=rule, result=result)
+        self._write_target_field(event=event, rule=rule, result=result)
 
     def _evaluate_cmdline(self, cmdline: str):
-        result = self._perform_misuse_detection(cmdline)
-        if result == 0:
-            return []
+        result = {}
 
-        return self._calculate_rule_attributions(cmdline)
+        malicious, result["confidence"] = self._perform_misuse_detection(cmdline)
+        if malicious:
+            result["attributions"] = self._calculate_rule_attributions(cmdline)
+
+        return result
 
     def _perform_misuse_detection(self, cmdline: str) -> int:
```

Comment:
```suggestion
    def _perform_misuse_detection(self, cmdline: str) -> Tuple[bool, float]:
```

The typing of the return of the method `_calculate_rule_attributions` below (line 241) is also not correct. It should be `List[dict]`.